unlock-app/src/components/creator/members/GrantKeysDrawer.tsx

```diff
@@ -0,0 +1,316 @@
+import React, { useEffect, useState, useContext } from 'react'
+import { useForm } from 'react-hook-form'
+import { Locks, Lock } from '../../../unlockTypes'
+import Drawer from '../../interface/Drawer'
+import { WalletServiceContext } from '../../../utils/withWalletService'
+import { Web3ServiceContext } from '../../../utils/withWeb3Service'
+import { AuthenticationContext } from '../../../contexts/AuthenticationContext'
+
+import {
+  Input,
+  Label,
+  Select,
+  TransactionPendingButton,
+  Button,
+} from '../../interface/checkout/FormStyles'
+import { ACCOUNT_REGEXP } from '../../../constants'
+import { getAddressForName } from '../../../hooks/useEns'
+import useAlert from '../../../hooks/useAlert'
+import Alert from '../../interface/Alert'
+
+interface GrantKeyFormProps {
+  lock: Lock
+  onGranted: (granted: boolean) => void
+}
+
+// Prevents re-rendering when time changes!
+const now = new Date().getTime()
+
```

Comment:
Can you please explain this part a little more? 

The way I see it is, this will only run once when the component file is loaded and stick to the same now. So this won't change regardless of the component mount and unmount. Am I correct? 

I feel this is an unnecessary optimization and could be made better.

Reply:
You are correct. The challenge is that if I put it in the lifecycle of the component then of course it changes all the time (and creates perpetual re-renders). The solution would be to put it in a `useEffect` hook but that seems pretty verbose for something not really that critical, because it in the end the user will pick an expiration in "days" more than seconds anyway? so the most precise time does not matter that much?